### PR TITLE
 show red background and warning symbol on project save error

### PR DESCRIPTION
**Pull request by davidsbailey**
spec:

red-error-bar

what you see on save error with this PR:

screen shot 2018-02-20 at 6 53 30 pm

screen shot 2018-02-20 at 6 52 51 pm

screen shot 2018-02-20 at 7 17 06 pm

### CODE REVIEW

**Comment by davidsbailey** [comment]
@islemaster friendly ping, in case this fell to the bottom of the morning email list

**Comment by islemaster** [comment]
It definitely did, sorry about that!  Looking now.

File: apps/src/code-studio/initApp/project.js [hunk 1]
```diff
@@ -431,6 +431,10 @@ var projects = module.exports = {
     }
   },
 
+  showHeaderSaveError() {
+    header.showProjectSaveError();
+  },
```
Comment (islemaster) [inline apps/src/code-studio/initApp/project.js]: Does this method add any value over calling `header.showProjectSaveError()` directly in `showSaveError_` below?

File: apps/src/code-studio/header.js [hunk 2]
```diff
@@ -502,4 +502,17 @@ header.updateTimestamp = function () {
   }
 };
 
+// TODO i18n
+header.showProjectSaveError = () => {
+  const saveErrorTooltip = "It looks like we couldn't save your progress. Make sure you have a " +
+    "good internet connection and try running the project again to save it.";
+
+  const saveErrorHtml = `
+<span class="save_error" title="${saveErrorTooltip}">
```
Comment (islemaster) [inline apps/src/code-studio/header.js]: Nit: [Our styleguide](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#css) says we follow [thoughtbot's SASS styleguide](https://github.com/thoughtbot/guides/tree/master/style/sass) for CSS, which says:

> Use hyphens when naming mixins, extends, classes, functions & variables: `span-columns` not `span_columns` or `spanColumns`.

I bet this is hard to enforce via linter, given how often classnames and such are dynamic or inline.

File: apps/src/code-studio/initApp/project.js (hunk 1 above)
Comment (davidsbailey) [reply apps/src/code-studio/initApp/project.js]: It seems we are grouping methods which call into the header, for possible future extraction. however it seems unlikely that will happen any time soon, so I'm ok with inlining.

File: apps/src/code-studio/header.js (hunk 2 above)
Comment (davidsbailey) [reply apps/src/code-studio/header.js]: ahh, that's good to know! I've always chosen arbitrarily.

**Pull request merged**